components/dashboard/src/components/WorkspaceLogs.tsx

```diff
@@ -54,17 +56,37 @@ export default function WorkspaceLogs(props: WorkspaceLogsProps) {
         terminal.loadAddon(fitAddon);
         terminal.open(xTermParentRef.current);
 
-        const logListener = (logs: string) => {
-            if (terminal && logs) {
-                terminal.write(logs);
+        let logBuffer = "";
+        let isWriting = false;
+
+        const processNextLog = () => {
+            if (isWriting || logBuffer.length === 0) return;
+
+            isWriting = true;
+            const logs = logBuffer.slice(0, MAX_CHUNK_SIZE);
+            logBuffer = logBuffer.slice(logs.length);
+            if (logs) {
```

Comment:
This feels like it's missing a:
```
if (!logs) {
  isWriting = false;
}
```
maybe as an else branch...?

Reply:
Fixed in https://github.com/gitpod-io/gitpod/pull/20065/commits/ff6a79f72ee29bc0eb64f191516c66c847bba4e3, thx!